### PR TITLE
Fix Tests, make sure all CompletionStages are complete before asserti…

### PR DESCRIPTION
**Pull request by billoneil**
…ons run.

I believe this will make sure the tests are deterministic.

### CODE REVIEW

File: src/test/java/org/jooq/lambda/SameExecutorCompletionStageTest.java [hunk 1]
```diff
@@ -39,17 +38,21 @@ public void setUp() {
...
         latch1.await();
+        firstCompletionStage.toCompletableFuture().join();
         assertHits(executorA, 1);
         
         CompletionStage<Void> secondCompletionStage = firstCompletionStage.thenApply(Function.identity());
+        secondCompletionStage.toCompletableFuture().join();
         assertHits(executorA, 1);
         
         CompletionStage<Void> thirdCompletionStage = secondCompletionStage.thenApplyAsync(latchIdentity(latch2));
+        thirdCompletionStage.toCompletableFuture().join();
```
Comment (billoneil) [inline src/test/java/org/jooq/lambda/SameExecutorCompletionStageTest.java]: Hmm this seems like a bug. This should have been below the await.

**Comment by billoneil** [comment]
@lukaseder let me know if you think this is looks correct.

**Comment by lukaseder** [comment]
This is definitely one of the reasons why I stayed clear of the concurrency APIs thus far... I'll trust you on this. Will review again later, before releasing.

**Comment by lukaseder** [comment]
And thanks again!

**Comment by lukaseder** [comment]
Good news from travis: https://travis-ci.org/jOOQ/jOOL/builds/205374593

**Pull request merged**